netlogo-gui/src/main/app/NotificationBanner.scala

```diff
@@ -0,0 +1,251 @@
+        return
+      }
+      val html = InfoFormatter.toInnerHtml(formattedString)
+
+      if (!jsonContent.trim.isEmpty) {
+        SwingUtilities.invokeLater(() => {
+          val editorPane: JEditorPane = new JEditorPane {
+            self =>
+            addFocusListener(new FocusListener {
+              def focusGained(fe: FocusEvent): Unit = {
+                FindDialog.watch(self)
+              }
+
+              def focusLost(fe: FocusEvent): Unit = {
+                if (!fe.isTemporary) FindDialog.dontWatch(self)
+              }
+            })
+            setDragEnabled(false)
+            setEditable(false)
+            setContentType("text/html")
+            setOpaque(false)
+            setText(html)
+            setCaretPosition(0)
+          }
+
+          val scrollPane = new JScrollPane(editorPane)
+          scrollPane.setPreferredSize(new Dimension(500, 400))
+          val panel = new JPanel(new BorderLayout())
+          panel.add(scrollPane, BorderLayout.CENTER)
+          val options: Array[AnyRef] = Array(I18N.gui.get("common.buttons.ok"))
+
+          val result = JOptionPane.showOptionDialog(
+            null, panel, I18N.gui.get("dialog.interface.newsNotificationTitle"),
+            JOptionPane.DEFAULT_OPTION, JOptionPane.PLAIN_MESSAGE, null, options, options(0)
+          )
+
+          // Check if OK was clicked (index 0 in options array) and hide the NotificationBanner
+          if (result == 0) {
+            setVisible(false) // Hide NotificationBanner
+            prefs.putInt("lastSeenEventId", jsonObjectList.head.eventId)
+          }
+        })
+      }
+    } catch {
+      case e: Exception => throw new Exception(s"Error in showJsonInDialog: ${e.getMessage}", e)
+    }
+  }
+
+
+  // Lazily initialize the Markdown parser and renderer (can be reused for multiple calls)
+  private lazy val markdownParser = Parser.builder().build()
+  private lazy val htmlRenderer = HtmlRenderer.builder().build()
+
+  def formatJsonObjectList(jsonObjectList: List[JsonObject]): String = {
+    jsonObjectList.map { obj =>
+      // Convert fullText from Markdown to HTML
+      val fullTextHtml = htmlRenderer.render(markdownParser.parse(obj.fullText))
+
+      // Format title, date, and the converted HTML fullText
+      s"<h3>* ${I18N.gui.get("dialog.interface.update")}: ${obj.date} -- ${obj.title}</h3>" +
+        s"$fullTextHtml"
+    }.mkString("<html>", "", "</html>")
+  }
+
+    // Static method to get the title of the head of jsonObjectList, with lazy initialization
+  def getJsonObjectHead: Option[String] = {
+    // Check if jsonObjectList is null or empty, and fetch the JSON if needed
+    if (jsonObjectList == null) {
+      val jsonContent = fetchJsonFromUrl(JsonUrl) // Use the static URL here
+      jsonObjectList = parseJsonToList(jsonContent)
+    }
+
+    // Return the title of the first item if jsonObjectList is not null and has elements
+    jsonObjectList match {
+      case null | Nil =>
```

Comment:
Ideally lists would never be `null`, since it's private you can make sure it's always `Nil` when it's empty.

Reply:
I did try using just Nil, but when I turn off my internet connection or point the url to some incorrect address, I needed to catch the null to gracefully handle the exception being caught

Reply:
`jsonObjectList` is only ever assigned using the result of your function, so you should be able to catch the internet issue in that function and return `Nil` there instead of `null`. For example, on line 105, you can return `Nil` there.

Reply:
Good catch about the internet issue, though.

Reply:
I just tried it with returning Nil rather than null in that line, and it threw an exception so I think I'll keep the null return unless anyone feels strongly:
nfo] jsonObjectList is empty or null; hiding NotificationBanner.
[info] Unable to connect to ccl.northwestern.edu but not throwing an exception
[error] NetLogo Startup Error
[error] java.util.NoSuchElementException: head of empty list
[error] 	at scala.collection.immutable.Nil$.head(List.scala:469)
[error] 	at scala.collection.immutable.Nil$.head(List.scala:466)
[error] 	at org.nlogo.app.NotificationBanner.isShowNeeded(NotificationBanner.scala:235)
[error] 	at org.nlogo.app.NotificationBanner.<init>(NotificationBanner.scala:40)
[error] 	at org.nlogo.app.App.<init>(App.scala:424)
[error] 	at java.base/jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
[error] 	at java.base/jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:77)
[error] 	at java.base/jdk.internal.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
[error] 	at java.base/java.lang.reflect.Constructor.newInstanceWithCaller(Constructor.java:500)
[error] 	at java.base/java.lang.reflect.Constructor.newInstance(Constructor.java:481)
[error] 	at org.picocontainer.injectors.AbstractInjector.newInstance(AbstractInjector.java:145)
[error] 	at org.picocontainer.injectors.ConstructorInjector$1.run(ConstructorInjector.java:342)
[error] 	at org.picocontainer.injectors.AbstractInjector$ThreadLocalCyclicDependencyGuard.observe(AbstractInjector.java:270)
[error] 	at org.picocontainer.injectors.ConstructorInjector.getComponentInstance(ConstructorInjector.java:364)
[error] 	at org.picocontainer.injectors.AbstractInjectionFactory$LifecycleAdapter.getComponentInstance(AbstractInjectionFactory.java:56)
[error] 	at org.picocontainer.behaviors.AbstractBehavior.getComponentInstance(AbstractBehavior.java:64)
[error] 	at org.picocontainer.behaviors.Stored.getComponentInstance(Stored.java:91)
[error] 	at org.picocontainer.DefaultPicoContainer.getInstance(DefaultPicoContainer.java:699)
[error] 	at org.picocontainer.DefaultPicoContainer.getComponent(DefaultPicoContainer.java:647)
[error] 	at org.picocontainer.DefaultPicoContainer.getComponent(DefaultPicoContainer.java:678)

Reply:
It threw that exception because on that line you're checking whether it's `null` instead of `Nil`, which would be an incorrect check if you were to return `Nil` from `parseJsonToList` instead of `null`. It's not a huge deal, I just personally feel it's more robust to keep empty lists as `Nil` rather than `null`. At the end of the day, if it works it works.